src/cupbearer/scripts/_shared.py

```diff
@@ -32,34 +39,43 @@ def __init__(
 
         if isinstance(model, HookedModel):
             self.model = model
-        elif input_shape is None:
+        elif input_format is None:
             raise ValueError(
-                "Must provide input_shape when passing a ModelConfig "
+                "Must provide input_format when passing a ModelConfig "
                 "instead of an instantiated model."
             )
         else:
-            self.model = model.build_model(input_shape=input_shape)
+            self.model = model.build_model(input_format=input_format)
         self.optim_cfg = optim_cfg
         self.val_loader_names = val_loader_names
         self.test_loader_names = test_loader_names
-        self.train_accuracy = Accuracy(task="multiclass", num_classes=num_classes)
+        self.task = task
+        self.loss_func = self._get_loss_func(self.task)
+        self.train_accuracy = Accuracy(
+            task=self.task, num_classes=num_classes, num_labels=num_labels
+        )
         self.val_accuracy = torch.nn.ModuleList(
             [
-                Accuracy(task="multiclass", num_classes=num_classes)
+                Accuracy(task=self.task, num_classes=num_classes, num_labels=num_labels)
                 for _ in val_loader_names
             ]
         )
         self.test_accuracy = torch.nn.ModuleList(
             [
-                Accuracy(task="multiclass", num_classes=num_classes)
+                Accuracy(task=self.task, num_classes=num_classes, num_labels=num_labels)
                 for _ in test_loader_names
             ]
         )
 
+    def _get_loss_func(self, task):
+        if self.task == torch_enums.ClassificationTask.MULTICLASS:
```

Comment:
```suggestion
        if self.task == "multiclass":
```
I think it would also be reasonable to use the `torchmetrics` enum everywhere (i.e. in the type annotation and default argument), but we should probably be consistent (and if we do use the enum, make sure we support all its possible values of course)